CONTRIBUTING.md

```diff
@@ -0,0 +1,39 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish to make via issue,
+email, or any other method with the owners of this repository before making a change. 
+
+Please note we have a code of conduct, please follow it in all your interactions with the project.
+
+## Issues Process
+
+1. First step while searching for issues is to check the ones with `core` and `high-priority`.
+2. Then `unimplemented` issues should be checked which got closed due to lack of attention but still need to be implemented.
+3. Then `deferred` issues may be valid now to take up.
+4. If no such criteria meets, only then move on to create new issues during an issue hunt. Of course issues about serious bugs and new ideas about features are always welcome.
+
+## Pull Request Process
+
+1. Ensure any install or build dependencies are removed before the end of the layer when doing a build.
```

Comment:
Change to `Ensure that the tests are passing locally by running ./gradlew build and  there are no checkstyle or PMD issues`